### ISSUE TITLE
Possible bug in RightGreenMultiplierNC

### ISSUE THREAD

**Issue by ZoltanCoccyx** [semigroups/Semigroups#929]
Hello!

The following code seems to me to be an example where the computed right green multiplier between d and h that are in the same RClass, and yet d*r is not even in D (and in particular not equal to h. I might have a left and right induced headache but I am almost sure this is not intended ^^.

Best,

Balthazar

```gap
LoadPackage("semigroups");

T := Monoid(Bipartition([[ 1, 2, -6 ], [ 3, 4, -1 ], [ 5, -2 ], [ 6 ], [ -3 ], [ -4 ], [ -5 ]]), 
Bipartition([[ 1, 2, 3, 5, -2, -6 ], [ 4, -5 ], [ 6, -3, -4 ], [ -1 ]]), 
Bipartition([[ 1, 2, 6, -1 ], [ 3, -3 ], [ 4, 5, -6 ], [ -2 ], [ -4, -5 ]]), 
Bipartition([[ 1, 3, -4 ], [ 2, 4, 5, -1, -5 ], [ 6, -2, -3, -6 ]]), 
Bipartition([[ 1, 3, 4, 6, -5 ], [ 2, 5, -1, -2, -4 ], [ -3, -6 ]]));
d := Bipartition([[ 1, 3, -2 ], [ 2, 4, 5, 6, -1, -6 ], [ -3 ], [ -4 ], [ -5 ]]);
D := DClasses(T)[17];
d := Representative(D);
H := HClasses(RClass(T,d))[3];
h := Representative(H);
r := RightGreensMultiplierNC(T, d, h);
d*r in D; # Should be true
```

**Comment by james-d-mitchell** [comment]
Thanks for the report, I'll investigate and get back to you!

**Comment by james-d-mitchell** [comment]
You're right this is a bug! I'm preparing a fix just now

**Comment by james-d-mitchell** [comment]
@ZoltanCoccyx the fix is in #930, I hope that resolves everything, if not keep the issues coming!

**Comment by ZoltanCoccyx** [comment]
I do not detect any issues related to that anymore, whether on transformations, partial permutations, or bipartitions :) Thanks !
